### PR TITLE
Update configuration api

### PR DESCRIPTION
**Pull request by shivamkm07**
# Description

This PR updates the config API quickstart to use V1 endpoint. It also updates the required sdk quickstarts as per sdk updates.

## Issue reference

We strive to have all PRs being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #832 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] The quickstart code compiles correctly
* [ ] You've tested new builds of the quickstart if you changed quickstart code
* [ ] You've updated the quickstart's README if necessary
* [ ] If you have changed the steps for a quickstart be sure that you have updated the automated validation accordingly. All of our quickstarts have annotations that allow them to be executed automatically as code. For more information see [mechanical-markdown](https://github.com/dapr/mechanical-markdown#readme). For user guide with examples see [Examples](https://github.com/dapr/mechanical-markdown/tree/main/examples#mechanical-markdown-by-example).

### CODE REVIEW

**Comment by shivamkm07** [comment]
@msfussell @paulyuk This PR updates config API quickstarts to use v1 endpoint. 

Also, there have been breaking changes in Go, Java and Python-sdks. I updated these sdk quickstarts to use the latest changes, but since the sdks haven't had releases yet, Python and Java validations are failing. (For Go, I updated go.mod to use latest master commit). Is there anyway we can use master version of SDKs in the workflow? Or should the sdk changes be in a separate PR later after the SDK releases?

**Review by msfussell — COMMENTED** [review]
Looks good. Need to add examples in here for:
1) C# client (only http currently)
2) Javascript client (only http currently)

**Comment by msfussell** [comment]
> @msfussell @paulyuk This PR updates config API quickstarts to use v1 endpoint.
> 
> Also, there have been breaking changes in Go, Java and Python-sdks. I updated these sdk quickstarts to use the latest changes, but since the sdks haven't had releases yet, Python and Java validations are failing. (For Go, I updated go.mod to use latest master commit). Is there anyway we can use master version of SDKs in the workflow? Or should the sdk changes be in a separate PR later after the SDK releases?

We need to get the RC versions included into the build. If pulling in Master is the way to go, it would be good to change the test for this. @paulyuk  for his opinion here

**Comment by paulyuk** [comment]
> > @msfussell @paulyuk This PR updates config API quickstarts to use v1 endpoint.
> > Also, there have been breaking changes in Go, Java and Python-sdks. I updated these sdk quickstarts to use the latest changes, but since the sdks haven't had releases yet, Python and Java validations are failing. (For Go, I updated go.mod to use latest master commit). Is there anyway we can use master version of SDKs in the workflow? Or should the sdk changes be in a separate PR later after the SDK releases?
> 
> We need to get the RC versions included into the build. If pulling in Master is the way to go, it would be good to change the test for this. @paulyuk for his opinion here

My plan is as follows:
- we keep master current and pinned with tagged releases or prereleases.  you can see now that master is already pinned to v1.11-rc2.  As rc3 comes out we'll change the pin.
- we begin the practice of pre-releases and releases of Quickstarts.  I made a 1.11-rc2 pre-release last night, and will do rc3 today.  The bulk of pre-release->release testing should focus *here*
- SDK Maintainers should coordinate with quickstart maintainers to tell us latest they want in the release.  Right now Configuration and Workflow are on my list since they appear to be broken.  This is a bit of a matrix with all the APIs x Languages so it relies on maintainers communicating well on needs.  otherwise we end up reactive like today.  

Thoughts?

**Comment by msfussell** [comment]
Sounds good to me. I posted on the Discord release channel to the SDK owners https://discord.com/channels/778680217417809931/804043781107613706/1111096250327126026

**Comment by shivamkm07** [comment]
> Looks good. Need to add examples in here for:
> 
> 1. C# client (only http currently)
> 2. Javascript client (only http currently)

We already have examples for C# and Javascript client(both http and sdk). It's just there were no changes in the sdk versions of these and so there is no change in those files. See https://github.com/dapr/quickstarts/tree/master/configuration/csharp/sdk, https://github.com/dapr/quickstarts/tree/master/configuration/javascript/sdk

File: configuration/go/sdk/order-processor/app.go [hunk 1]
```diff
@@ -12,37 +12,29 @@ import (
...
 	}
-	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
 
 	// Get config items from config store
-	for _, item := range CONFIGURATION_ITEMS {
-		config, err := client.GetConfigurationItem(ctx, DAPR_CONFIGURATION_STORE, item)
+	for _, key := range CONFIGURATION_KEYS {
+		config, err := client.GetConfigurationItem(ctx, DAPR_CONFIGURATION_STORE, key)
```
Comment (paulyuk) [inline configuration/go/sdk/order-processor/app.go:27]: note breaking change from item to key

**Review by paulyuk — APPROVED** [review]
LGTM - please update release notes so people understand API change

**Pull request merged**